daemon/containerd/image_exporter.go

```diff
@@ -342,3 +378,62 @@ func (i *ImageService) LoadImage(ctx context.Context, inTar io.ReadCloser, platf
 
 	return nil
 }
+
+// verifyImagesProvidePlatform checks if the requested platform is loaded.
+// If the requested platform is not loaded, it returns an error.
+func (i *ImageService) verifyImagesProvidePlatform(ctx context.Context, imgs []containerdimages.Image, platform ocispec.Platform, pm platforms.Matcher) error {
+	var imageNames []string
+	for _, img := range imgs {
+		hasRequestedPlatform := false
+		err := i.walkImageManifests(ctx, img, func(platformImg *ImageManifest) error {
+			imgPlat, err := platformImg.ImagePlatform(ctx)
+			if err != nil {
+				if cerrdefs.IsNotFound(err) {
+					return nil
+				}
+				return errors.Wrapf(err, "failed to determine image platform")
+			}
+
+			if pm.Match(imgPlat) {
+				available, err := platformImg.CheckContentAvailable(ctx)
+				if err != nil {
+					return errors.Wrapf(err,
+						"failed to determine image content availability for platform %s",
+						platforms.FormatAll(platform),
+					)
+				}
+
+				if available {
+					hasRequestedPlatform = true
+					return nil
+				}
+			}
+			return nil
+		})
+		if err != nil {
+			return errdefs.System(err)
+		}
+		if !hasRequestedPlatform {
+			imageNames = append(imageNames, imageFamiliarName(img))
+		}
+	}
+
+	allImageNames := strings.Join(imageNames, ", ")
+	platformStr := platforms.FormatAll(platform)
+
+	msg := ""
+	switch len(imageNames) {
+	case 0:
+	case 1:
+		msg = "image %s was loaded, but doesn't provide the requested platform (%s)"
+	default:
+		msg = "images [%s] were loaded, but don't provide the requested platform (%s)"
+
+	}
+
+	if msg != "" {
+		return errdefs.NotFound(fmt.Errorf(msg, allImageNames, platformStr))
+	}
```

Comment:
Perhaps it's more readable to avoid the extra indirect of creating a message only to change it into an error;


```go
	switch len(imageNames) {
	case 0:
		// success; none of the images are missing the requested platform.
	case 1:
		return errdefs.NotFound(fmt.Errorf("image %s was loaded, but doesn't provide the requested platform (%s)", allImageNames, platformStr))
	default:
		return errdefs.NotFound(fmt.Errorf("images [%s] were loaded, but don't provide the requested platform (%s)", allImageNames, platformStr))
	}
```